yjit/src/codegen.rs

```diff
@@ -5063,23 +5063,25 @@ unsafe extern "C" fn build_kwhash(ci: *const rb_callinfo, sp: *const VALUE) -> V
 // at sp[-2]. Depending on the frame type, it can serve different purposes,
 // which are covered here by enum variants.
 enum SpecVal {
-    None,
     BlockHandler(BlockHandler),
-    BlockParamProxy,
     PrevEP(*const VALUE),
     PrevEPOpnd(Opnd),
-    // To avoid holding values across C calls for complex calls,
-    // we might need to set the SpecVal earlier in the call sequence
-    AlreadySet,
 }
 
 // Each variant represents a branch in vm_caller_setup_arg_block.
 #[derive(Clone, Copy)]
 pub enum BlockHandler {
+    // passing no block
+    None,
```

Comment:
We are using `Option<BlockHandler>` to propagate a `block_handler`, but I think it should be changed to just `BlockHandler` if we add `None` variant to it. `None` and `Some(BlockHandler::None)` seem to mean the same thing.